command/agent.go

```diff
@@ -593,6 +594,9 @@ func (c *AgentCommand) Run(args []string) int {
 	}()
 
 	select {
+	case err := <-errCh:
+		c.UI.Error(fmt.Sprintf("Error encountered: %s", err))
```

Comment:
I think it's a mistake to emit runtime errors (as opposed to startup errors) this way.  It makes sense to use UI when errors occur at startup, e.g. bad arguments or config, because typically that happens when a user is experimenting manually.  Runtime errors, on the other hand, will normally not be encountered by the user.  They should go into a log, formatted as a log entry, and be tagged as errors such that log scrapers and aggregators will report them appropriately.

Reply:
That's fair. It was suggested that error emitted this way to be logged as a UI error, and I did observe this pattern on non-zero exit code runs. Would you suggest to still return a generic error in the UI output?

Reply:
I'm okay with that I guess, though I think it sets a bad precedent.  If we start logging *and* emitting UI errors here, then subsequently whenever a dev wants to log an error they're going to ask themselves whether they also need to do a UI output.

Reply:
Updated in child PR #9761, lmk what you think.